### PR TITLE
[WIP] Feature/roc 2506 store sealed claim form in em

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-2506

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: document-client/src/main/java/uk/gov/hmcts/document/configuration/DocumentManagementClientConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Value("${http.connect.request.timeout}")
+    private int httpConnectRequestTimeout;
+
+    @Value("${http.connect.socket.timeout}")
+    private int httpConnectSocketTimeout;
+
+    @Bean
+    public RestTemplate restTemplate() {
+        objectMapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
```
Comment (timja-kainos) [inline document-client/src/main/java/uk/gov/hmcts/document/configuration/DocumentManagementClientConfiguration.java]: We already configure an object mapper, you shouldn't need to configure your own one

File: document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementUploadService.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    @CircuitBreaker(exclude = HttpClientErrorException.class)
+    public List<FileUploadResponse> uploadFiles(List<MultipartFile> files, String authorizationToken, String requestId) {
+        MultiValueMap<String, Object> parameters = prepareRequest(files);
+
+        HttpHeaders httpHeaders = setHttpHeaders(authorizationToken);
+
+        HttpEntity<MultiValueMap<String, Object>> httpEntity = new HttpEntity<>(parameters,
+            httpHeaders);
+
+        JsonNode filesJsonArray = template.postForObject(documentManagementServiceURL, httpEntity, ObjectNode.class)
```
Comment (timja-kainos) [inline document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementUploadService.java]: we should be using document management api here not manually doing the http like in IdamApi and the BankHolidaysApi

File: document-client/src/test/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,25 @@
+
+spring.http.multipart.max-file-size=30000KB
```
Comment (timja-kainos) [inline document-client/src/test/resources/application.properties]: use yaml file instead, for consistency?

File: document-client/build.gradle [hunk 4]
```diff
@@ -0,0 +1,42 @@
+buildscript {
```
Comment (timja-kainos) [inline document-client/build.gradle]: this client should probably be in its own repo and published to jcenter

File: document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementDownloadService.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+@Service
+public class DocumentManagementDownloadService {
+
+    private static final Logger log = LoggerFactory.getLogger(DocumentManagementDownloadService.class);
+    private final static MediaType MEDIA_TYPE =
+        new MediaType("application", "vnd.uk.gov.hmcts.dm.document.v1+json");
+
+    @Value("${document.management.upload.file.url}")
+    private String evidenceManagementServiceURL;
```
Comment (timja-kainos) [inline document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementDownloadService.java]: documentManagementServiceUrL

File: document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementUploadService.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+            .get("_embedded")
+            .get("documents");
+
+        log.info("For Request Id {} : File upload response from Evidence Management service is {}", requestId,
```
Comment (timja-kainos) [inline document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementUploadService.java]: Document Management

File: document-client/src/test/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+auth.provider.service.client.key=AAAAAAAAAAAAAAAA
+
+auth.provider.service.client.tokenTimeToLiveInSeconds=900
+
+auth.provider.health.path=/health
+
+auth.provider.health.url=${auth.provider.service.client.baseUrl}${auth.provider.health.path}
+
+####Document management Service URL's
+document.management.upload.file.url=${EVIDENCE_MANAGEMENT_UPLOAD_FILE_URL:http://localhost:3404/documents}
```
Comment (timja-kainos) [inline document-client/src/test/resources/application.properties]: DOCUMENT

File: document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementDownloadService.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (sabahirfan) [reply document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementDownloadService.java]: Done

File: document-client/src/test/resources/application.properties (hunk 7 above)
Comment (sabahirfan) [reply document-client/src/test/resources/application.properties]: done

File: document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementUploadService.java (hunk 6 above)
Comment (sabahirfan) [reply document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementUploadService.java]: done

File: document-client/src/test/resources/application.properties (hunk 3 above)
Comment (sabahirfan) [reply document-client/src/test/resources/application.properties]: done

File: document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementDownloadService.java (hunk 5 above)
Comment (sabahirfan) [reply document-client/src/main/java/uk/gov/hmcts/document/service/DocumentManagementDownloadService.java]: done

**Comment by timja-kainos** [comment]
superseded by https://github.com/hmcts/cmc-claim-store/pull/80

**Pull request closed without merging**